src/TurbulenceClosures/turbulence_closure_implementations/convective_adjustment_vertical_diffusivity.jl

```diff
@@ -0,0 +1,220 @@
+
+* `background_κz` : Vertical tracer diffusivity in regions with stable buoyancy gradients.
+
+* `convective_νz` : Vertical viscosity in regions with unstable buoyancy gradients. Either
+                    a number, function, array, or field.
+
+* `background_κz` : Vertical viscosity in regions with stable buoyancy gradients.
+
+* `time_discretization` : Either `ExplicitTimeDiscretization` or `VerticallyImplicitTimeDiscretization`.
+"""
+function ConvectiveAdjustmentVerticalDiffusivity(FT=Float64;
+                                                 convective_κz = zero(FT),
+                                                 convective_νz = zero(FT),
+                                                 background_κz = zero(FT),
+                                                 background_νz = zero(FT),
+                                                 time_discretization::TD = VerticallyImplicitTimeDiscretization()) where TD
+
+    return ConvectiveAdjustmentVerticalDiffusivity{TD}(convective_κz, convective_νz,
+                                                       background_κz, background_νz)
+end
+
+const CAVD = ConvectiveAdjustmentVerticalDiffusivity
+
+#####
+##### Diffusivity field utilities
+#####
+
+function with_tracers(tracers, closure::ConvectiveAdjustmentVerticalDiffusivity{TD}) where TD
+    background_κz = tracer_diffusivities(tracers, closure.background_κz)
+    convective_κz = tracer_diffusivities(tracers, closure.convective_κz)
+    return ConvectiveAdjustmentVerticalDiffusivity{TD}(convective_κz,
+                                                       closure.convective_νz,
+                                                       background_κz,
+                                                       closure.background_νz)
+end
+
+
+function DiffusivityFields(arch, grid, tracer_names, bcs, closure::CAVD)
+    data = new_data(Bool, arch, grid, (Center, Center, Face))
+    stable_buoyancy_gradient = Field(Center, Center, Face, arch, grid, nothing, data)
+    return (; stable_buoyancy_gradient)
+end       
+
+function calculate_diffusivities!(diffusivities, arch, grid, closure::CAVD, buoyancy, velocities, tracers)
+
+    event = launch!(arch, grid, :xyz,
+                    compute_stability!, diffusivities, grid, tracers, buoyancy,
+                    dependencies = device_event(arch))
+
+    wait(device(arch), event)
+
+    return nothing
+end
+
+@inline is_stableᶜᶜᶠ(i, j, k, grid, tracers, buoyancy) = ∂z_b(i, j, k, grid, buoyancy, tracers) > 0
+
+@kernel function compute_stability!(diffusivities, grid, tracers, buoyancy)
+    i, j, k, = @index(Global, NTuple)
+    @inbounds  diffusivities.stable_buoyancy_gradient[i, j, k] = is_stableᶜᶜᶠ(i, j, k, grid, tracers, buoyancy)
+end
+
+#####
+##### Fluxes
+#####
+
+# u
+@inline function νᶠᶜᶠ(i, j, k, grid, clock, closure::CAVD, stable_buoyancy_gradient)
+    @inbounds stableᶠᶜᶠ = stable_buoyancy_gradient[i, j, k] || stable_buoyancy_gradient[i-1, j, k]
```

Comment:
I don't understand this line, why either `i` or `i-1` and not also `i+1`?
probably something that has to do with staggered grid details that I'm missing..

Reply:
The buoyancy gradient is defined at `Center, Center, Face` (superscript ccf). Thus we determine that the node fcf at Face, Center, Face is stable by checking the stability of the buoyancy gradient on either side of the node. By convention, face indices lie to the "left" of centered indices. Thus the buoyancy gradients at i and i-1 "span" the node at fcf.

Reply:
OK, makes sense. I was missing this "convention".

Reply:
This is a modeling choice --- I'm not aware that such an implementation of convective adjustment is standard or not. In fact, most convective adjustment schemes may not diffuse the velocity field (only tracers).

Reply:
I meant the convention regarding "face indices lie to the "left" of centered indices".